crates/runner/src/vm/instructions/store.rs

```diff
@@ -0,0 +1,446 @@
+//! STORE instructions for the Cairo M VM.
+//!
+//! STORE instructions are used to store values in the memory.
+//! They are used to store the result of an operation or the value of a variable.
+
+use crate::{
+    memory::{Memory, MemoryError},
+    vm::{instructions::Instruction, State},
+};
+
+/// Store the sum of the values at the offsets `fp + off0` and `fp + off1` in the memory at the offset `fp + off2`.
+///
+/// OPCODE ID: 0
+///
+/// CASM equivalent:
+/// ```casm
+/// [fp + off2] = [fp + off0] + [fp + off1]
+/// ```
+pub fn store_add_fp_fp(
+    memory: &mut Memory,
+    state: State,
+    instruction: Instruction,
+) -> Result<State, MemoryError> {
+    let [off0, off1, off2] = instruction.args;
+    let value = memory.get_data(state.fp + off0)? + memory.get_data(state.fp + off1)?;
+    memory.insert(state.fp + off2, value.into())?;
+
+    Ok(state.advance())
+}
+
+/// Store the sum of the value at the offset `fp + off0` and the immediate value `off1` in the memory at the offset `fp + off2`.
+///
+/// OPCODE ID: 1
+///
+/// CASM equivalent:
+/// ```casm
+/// [fp + off2] = [fp + off0] + off1
+/// ```
+pub fn store_add_fp_imm(
+    memory: &mut Memory,
+    state: State,
+    instruction: Instruction,
+) -> Result<State, MemoryError> {
+    let [off0, off1, off2] = instruction.args;
+    let value = memory.get_data(state.fp + off0)? + off1;
+    memory.insert(state.fp + off2, value.into())?;
+
+    Ok(state.advance())
+}
```

Comment:
What do you think about this below ?
I find it more clear to rename off1 to imm
If this is preferable, it should be applied to all instructions with an immediate value
```suggestion
/// Store the sum of the value at the offset `fp + off0` and the immediate value `imm` in the memory at the offset `fp + off2`.
///
/// OPCODE ID: 1
///
/// CASM equivalent:
/// ```casm
/// [fp + off2] = [fp + off0] + imm
/// ```
pub fn store_add_fp_imm(
    memory: &mut Memory,
    state: State,
    instruction: Instruction,
) -> Result<State, MemoryError> {
    let [off0, imm, off2] = instruction.args;
    let value = memory.get_data(state.fp + off0)? + imm;
    memory.insert(state.fp + off2, value.into())?;

    Ok(state.advance())
}
```